src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/NotificationOption2.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+#if CODE_STYLE
+using WorkspacesResources = Microsoft.CodeAnalysis.CodeStyleResources;
+#else
+using System.Security;
+using Roslyn.Utilities;
+#endif
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    /// <summary>
+    /// Offers different notification styles for enforcing
+    /// a code style. Under the hood, it simply maps to <see cref="DiagnosticSeverity"/>
+    /// </summary>
+    /// <remarks>
+    /// This also supports various properties for databinding.
+    /// </remarks>
+    /// <completionlist cref="NotificationOption2"/>
+    internal class NotificationOption2
+    {
+        public string Name { get; set; }
+
+        public ReportDiagnostic Severity
+        {
+            get;
+            set;
```

Comment:
```suggestion
```

Reply:
Tracked in https://github.com/dotnet/roslyn/issues/42550